kernel/src/main/java/org/kframework/compile/GenerateSortProjections.java

```diff
@@ -28,13 +28,16 @@
 public class GenerateSortProjections {
 
   private Module mod;
+  private final Module mainMod;
   private final boolean cover;
 
-  public GenerateSortProjections(boolean cover) {
+  public GenerateSortProjections(boolean cover, Module mainMod) {
+    this.mainMod = mainMod;
```

Comment:
Not setting `this.mod` in the constructor here is bug-prone. The `gen` overloads all assume `this.mod` is non-null, but not all them actually initialize it, e.g., `GenerateSortProjections(false, mainMod).gen(mySort)` would fail.

A better approach might be to remove `this.mod` and have every `gen` overload take an explicit `mod` argument.

Reply:
Ah just realized this was already an issue prior to your change. Feel free to ignore and I'll fix it in a follow up PR.